src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaSubmissionSendingService.java

```diff
@@ -0,0 +1,133 @@
+public class AthenaSubmissionSendingService {
+
+    private static final int SUBMISSIONS_PER_REQUEST = 100;
+
+    private final Logger log = LoggerFactory.getLogger(AthenaSubmissionSendingService.class);
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
+
+    private final TextSubmissionRepository textSubmissionRepository;
+
+    private final AthenaConnector<RequestDTO, ResponseDTO> connector;
+
+    /**
+     * Creates a new AthenaSubmissionSendingService.
+     *
+     * @param athenaRestTemplate       the RestTemplate to use for requests to the Athena service
+     * @param textSubmissionRepository the repository to use for finding submissions in the database
+     */
+    public AthenaSubmissionSendingService(@Qualifier("athenaRestTemplate") RestTemplate athenaRestTemplate, TextSubmissionRepository textSubmissionRepository) {
+        this.textSubmissionRepository = textSubmissionRepository;
+        connector = new AthenaConnector<>(log, athenaRestTemplate, ResponseDTO.class);
+    }
+
+    private static class RequestDTO {
+
+        public TextExerciseDTO exercise;
+
+        public List<TextSubmissionDTO> submissions;
+
+        RequestDTO(@NotNull TextExercise exercise, @NotNull Set<TextSubmission> submissions) {
+            this.exercise = TextExerciseDTO.of(exercise);
+            this.submissions = submissions.stream().map(submission -> TextSubmissionDTO.of(exercise.getId(), submission)).toList();
+        }
+    }
+
+    private record ResponseDTO(String data) {
+    }
+
+    /**
+     * Calls the remote Athena service to submit a job for calculating automatic feedback
+     *
+     * @param exercise the exercise the automatic assessments should be calculated for
+     */
+    public void sendSubmissions(TextExercise exercise) {
+        sendSubmissions(exercise, 1);
+    }
+
+    /**
+     * Calls the remote Athena service to submit a Job for calculating automatic feedback
+     * Note: See `TextSubmissionService:getTextSubmissionsByExerciseId` for selection of Submissions.
+     *
+     * @param exercise   the exercise the automatic assessments should be calculated for
+     * @param maxRetries number of retries before the request will be canceled
+     */
+    public void sendSubmissions(TextExercise exercise, int maxRetries) {
+        if (!exercise.isFeedbackSuggestionsEnabled()) {
+            throw new IllegalArgumentException("The Exercise does not have feedback suggestions enabled.");
+        }
+
+        log.debug("Start Athena Submission Sending Service for Text Exercise '{}' (#{}).", exercise.getTitle(), exercise.getId());
+
+        // Find all text submissions for exercise (later we will support others)
+        int page = 0;
+        while (true) {
+            Pageable pageRequest = PageRequest.of(page, SUBMISSIONS_PER_REQUEST);
+            Page<TextSubmission> textSubmissions = textSubmissionRepository.findByParticipation_ExerciseIdAndSubmittedIsTrue(exercise.getId(), pageRequest);
```

Comment:
You can use `nextPageable()` so you don't need to keep track of the index by yourself

Reply:
I'm now using `pageRequest.next` 👍